[browserperfdash-benchmark] Fix error logging from server after 251530@main

clopez opened the pull request:
#### c0299cc3d772e526b3988562d2ec8ebe341b5787
<pre>
[browserperfdash-benchmark] Fix error logging from server after 251530@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=242242">https://bugs.webkit.org/show_bug.cgi?id=242242</a>

Reviewed by Adrian Perez de Castro.

After porting to Python3 on 251530@main (r295525) the script not longer logs
in stderr the reply from the server when this replies with an error. It only
logs a generic exception.

This is caused because urllib.request.urlopen on Python3 always raises an
exception when the http server reply is different from 200.
This patch wraps the call to return the HTTPError object instead of raising
the exception. This HTTPError object can be used later as an HTTPResponse one,
so we can read the reply from the server and log it.

* Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py:
(BrowserPerfDashRunner):
(BrowserPerfDashRunner._send_post_request_data):
(BrowserPerfDashRunner._upload_result):

Canonical link: <a href="https://commits.webkit.org/252040@main">https://commits.webkit.org/252040@main</a>
</pre>

aperezdc commented on Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py:
```diff
@@ -103,6 +103,15 @@ def _get_test_data_json_string(self, temp_result_file):
             del temp_result_json['debugOutput']
         return json.dumps(temp_result_json)
 
+    # urllib.request.urlopen always rises an exception when the http return code is not 200
```
Typo: `s/rises/raises/`

```suggestion
    # urllib.request.urlopen always raises an exception when the http return code is not 200
```

aperezdc commented on Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py:
```diff
@@ -103,6 +103,15 @@ def _get_test_data_json_string(self, temp_result_file):
 
+    # urllib.request.urlopen always rises an exception when the http return code is not 200
+    # so this wraps the call to return the HTTPError object instead of rising the exception.
+    # The HTTPError object can can be treated later as a http.client.HTTPResponse object
```
No need to dance can-can here, one “can” 🐶 is enough:

 ```suggestion
    # The HTTPError object can be treated later as a http.client.HTTPResponse object
```

aperezdc reviewed (approved):
@clopez  r=me with nits for a couple of types in the comments; please take a look at them before landing.

clopez committed:
[browserperfdash-benchmark] Fix error logging from server after 251530@main
https://bugs.webkit.org/show_bug.cgi?id=242242

Reviewed by Adrian Perez de Castro.

After porting to Python3 on 251530@main (r295525) the script not longer logs
in stderr the reply from the server when this replies with an error. It only
logs a generic exception.

This is caused because urllib.request.urlopen on Python3 always raises an
exception when the http server reply is different from 200.
This patch wraps the call to return the HTTPError object instead of raising
the exception. This HTTPError object can be used later as an HTTPResponse one,
so we can read the reply from the server and log it.

* Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py:
(BrowserPerfDashRunner):
(BrowserPerfDashRunner._send_post_request_data):
(BrowserPerfDashRunner._upload_result):

Canonical link: https://commits.webkit.org/252040@main
